src/gbcc_video.c

```diff
@@ -270,24 +276,53 @@ void gbcc_set_video_mode(struct gbc *gbc, uint8_t mode)
 	gbcc_memory_write(gbc, STAT, stat, true);
 }
 
-uint32_t get_palette_colour(uint8_t palette, uint8_t n)
+uint32_t get_palette_colour(uint8_t palette, uint8_t n, enum palette_flag pf)
 {
 	uint8_t colours[4] = {
 		(palette & 0x03u) >> 0u,
 		(palette & 0x0Cu) >> 2u,
 		(palette & 0x30u) >> 4u,
 		(palette & 0xC0u) >> 6u
 	};
-	switch (colours[n]) {
-		case 3:
-			return 0x000000u;
-		case 2:
-			return 0x6b7353u;
-		case 1:
-			return 0x8b956du;
-		case 0:
-			return 0xc4cfa1u;
-		default:
-			return 0;
+	switch (pf) {
```

Comment:
A nested switch is messier than it could be - a selection of palette arrays for example would be much neater:
```c
uint32_t red_palette = [0x000000u, 0x7f3848u, 0xff8096u, 0xf8f8f8u];
...
switch (pf) {
        case BACKGROUND:
                return red_palette[colours[n]];
        case SPRITE_1:
                ...
}
```